### PR TITLE
Tabulated command suggestions, and tried to improve readability here and...

### PR DESCRIPTION
**Pull request by SmartViking**
... there

I'm trying to get familiar with the codebase by reading (with an eye towards implementing recursive alias evaluation #109), and I might as well try to improve some things here and there in the process.

The changes I make pertaining to readability are based on the supposition that expression-oriented code is more readable than statement-oriented code (i.e. imperative vs functional). Let me know if you disagree on that.

### CODE REVIEW

File: xonsh/built_ins.py [hunk 1]
```diff
@@ -140,13 +140,11 @@ def get(self, key, default=None):
...
-            val = builtins.evalx(val, glbs=ctx, locs=ctx)
+            return builtins.evalx(val, glbs=ctx, locs=ctx)
         elif isinstance(val, Iterable) or callable(val):
-            pass
+            return val
         else:
-            msg = 'alias of {0!r} has an inappropriate type: {1!r}'
-            raise TypeError(msg.format(key, val))
-        return val
+            raise TypeError('alias of {!r} has an inappropriate type: {!r}'.format(key, val))
```
Comment (scopatz) [inline xonsh/built_ins.py]: This line is over 80 chars which is why I put the msg on the previous line.

File: xonsh/tools.py [hunk 2]
```diff
@@ -211,34 +211,38 @@ class redirect_stderr(_RedirectStream):
...
-            out.append('Did you mean the following?')
-        for i in range(num):
-            k,v = suggested.popitem(False)
-            out.append('    {0}: {1}'.format(k,v))
-        return '\n'.join(out)
+        
+        if num == 0:
+            return ''
+        else:
+            tips = 'Did you mean {}the following?'.format('' if num == 1 else 'one of ')
```
Comment (scopatz) [inline xonsh/tools.py]: This line is also over 80 chars.

**Comment by scopatz** [comment]
Thanks for putting this in @SmartViking! It is much agreed. There are some small PEP8 issues with regards to the line length, but otherwise, great.

> The changes I make pertaining to readability are based on the supposition that expression-oriented code is more readable than statement-oriented code (i.e. imperative vs functional). Let me know if you disagree on that.

I don't know that I agree or disagree with that supposition.  I think each has their place and what is nice about Python is that you can choose which is most appropriate for any given expression. I think both modes can be made readable or unreadable.  So if two expressions perform the same task and have equal readability, then the main metric to judge them on is performance.  So I am not sure it matters too much.

**Comment by scopatz** [comment]
Hi @SmartViking - it looks like this now has some conflicts with master.  Sorry!  Could you resolve them and then ping me here, please? Thanks!

**Comment by SmartViking** [comment]
Ping!

**Comment by scopatz** [comment]
Thanks @SmartViking!

**Pull request merged**